Reifier.fs

```diff
@@ -12,54 +12,249 @@ open Starling.Core.Model
 open Starling.Core.Command
 open Starling.Core.GuardedView
 open Starling.Core.Symbolic
+open Starling.TermGen.Iter
 
+/// Splits an iterated GFunc into a pair of guard and iterated func.
+let iterGFuncTuple
+  ({ Iterator = i; Func = { Cond = c; Item = f }} : IteratedGFunc<'Var>)
+  : BoolExpr<'Var> * IteratedContainer<Func<Expr<'Var>>, IntExpr<'Var>> =
+    (c, iterated f i)
+
+/// <summary>
+///     Generates the powerset of a set expressed as a list.
+/// </summary>
+/// <param name="xs">The set whose powerset is sought.</param>
+/// <typeparam name="A">The type of set elements.</typeparam>
+/// <returns>
+///     The powerset of <paramref name="xs"/>, as a lazy sequence.
+/// </returns>
+let rec powerset (xs : 'A list) : 'A list seq =
+    // originally by Tomas Petricek: see http://stackoverflow.com/a/16826554
+    seq {
+        match xs with
+        | [] -> yield []
+        | h::t -> for x in powerset t do yield! [x; h::x] }
+
+/// <summary>
+///     Given a series of iterated funcs
+///     <c>A(a)[i] * A(b)[j] * A(c)[k] * ...</c>, generate the equalities
+///     <c>i==j, i==k, ...</c>.
+/// </summary>
+/// <param name="flist">The NON-EMPTY list of funcs to generate over.</param>
+/// <returns>The series of parameter equalities described above.</returns>
+let paramEqualities (flist : IteratedFunc<Sym<MarkedVar>> list)
+  : BoolExpr<Sym<MarkedVar>> list =
+    let x = List.head flist
+    concatMap
+        (fun y -> List.map2 mkEq x.Func.Params y.Func.Params)
+        (List.tail flist)
+
+/// <summary>
+///     Preprocesses a view for reification.
+///
+///     <para>
+///         This function converts a view multiset into a list, and expands out
+///         any case splits over potentially-equal iterated funcs:
+///         for example, <c>(G1 -> A(x)[i]) * (G2 -> A(y)[j])</c> will also
+///         generate the func <c>(G1 ^ G2 ^ x=y -> A(x)[i+j])</c> when
+///         A(x) is iterated.
+///     </para>
+/// </summary>
+/// <param name="protos">The prototypes used to find iterated funcs.</param>
+/// <param name="view">The view to preprocess.</param>
+/// <returns>The preprocessed view, as a func list.</returns>
+let preprocessView
```

Comment:
Not sure exactly what would be useful here.  Have added a bit more verbiage but I'm not sure if it helps or hinders.